articles/140_topic_and_service_name_mapping.md

```diff
@@ -219,13 +201,94 @@ Since ROS 2 topic and service names are expanded to fully qualified names, any b
+All this stays generally ROS agnostic and open to alternative implementations of the RMW interface (e.g. ZeroMQ etc.), which may want to introduce another, different namespace system.
+
+#### DDS Partitions Characteristics
+
+A DDS topic at this point is the basename token within a ROS topic.
+The namespace of a ROS topic is implemented within the partition field of DDS.
+DDS partitions are implemented as a string array within the Subscription/Publisher QoS, allowing up to 64 items.
+The maximum size of each string has yet to be verified and may depend strongly on the implementation.
+Hereby, the partitions field is implemented as an array of strings, where each string is an individual partitions.
+This means further that every partition index is stricly unique and independent from all other items.
+Special character used within regular expressions (such as `+`, `*`, `^`) are officially not supported by the standard.
+
+DDS Partitions don't support a partition hierarchy.
+Each entry in the partition array is directly combined with the DDS topic and they are not sequentially combined.
+If a publisher operates on two partition entries, e.g. `foo` and `bar` with a base name of `baz`, the resulting topic will be twofold:
+
+```
+- /foo/baz
+- /bar/baz
+```
+
+That implies, that DDS Partitions by default are not depicting any hierarchy.
+On the other hand, it allows an easy aliasing or duplication of the specified topic, given that the basename topic does not change.
+A dynamic duplication of the original topic such as `cmd_vel` in a namespace `robot1` to `robot2/cmd_vel` can thus be easily achieved by expanding the DDS partition array.
+
+#### Hierarchy with DDS Partitions
+
+Given that DDS Partitions are not hierarchical, we have to setup the hierarchy on top.
+For this, we use the forward slash `/` explicitely in each partition field.
+If we want to replicate a hierarchy of `warehouse`, `robot1`, `camera_left`, and finally the basename `image_raw`, the first entry of the partitions would look like:
+
+```
+warehouse/robot1/camera_left
+```
+
+#### Remapping with DDS Partitions
+
+We have to differentiate remapping from aliasing at this point.
+As described before, aliasing implies a duplication of a topic.
+Remapping, in contrast, alters an existing topic and thus remaps one topic into another.
+Given the nature of DDS Partitions, aliasing is natively supported by adding a second entry in the partition field.
+Remapping means, we have to change one existing partition field.
+However, as the complete hierarchy is illustrated within the first index of the partitions, the act of remapping resolves into a string modification.
+If we want to remap a camera image from namespace `camera1` to namespace `camera2`, we have to modify the string in the respective field.
+Publishers or Subscribers are uneffected by this change and don't have to be shutdown and restarted.
```

Comment:
I am not sure that we should make this guarantee. Depending on the rmw impl. this might be impossible to do without closing the existing publisher and creating a new one.